jina/clients/base.py

```diff
@@ -161,7 +161,7 @@ async def _get_results(
                 with cm1 as p_bar, cm2:
                     async for resp in stub.Call(req_iter):
                         resp.as_typed_request(resp.request_type)
-                        resp.as_response()
+                        resp = resp.as_response()
```

Comment:
This feels like the new `resp` should be a completely new object and not just a week reference. I smell problems here in the future. Anyhow, good for now.

Reply:
For performance reasons, all these `types` objects are views to the same `proto` object